### PR TITLE
adding DQM to GeneralTracks for Muon Alignment

### PR DESCRIPTION
**Pull request by lpernie**

### CODE REVIEW

**Comment by lpernie** [comment]
@mmusich done!

**Comment by ghellwig** [comment]
please test

**Comment by mmusich** [comment]
@ghellwig this PR is not in a state to be tested yet...

**Comment by mmusich** [comment]
local tests with `runTheMatrix.py -l limited` yield:
```
4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC Step0-PASSED Step1-FAILED Step2-NOTRUN Step3-NOTRUN Step4-NOTRUN  - time date Wed Sep  7 09:50:56 2016-date Wed Sep  7 09:50:24 2016; exit: 0 256 0 0 0
5.1_TTbar+TTbarFS+HARVESTFS Step0-PASSED Step1-PASSED  - time date Wed Sep  7 09:56:24 2016-date Wed Sep  7 09:50:26 2016; exit: 0 0
8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS Step0-PASSED Step1-PASSED Step2-FAILED Step3-NOTRUN Step4-NOTRUN  - time date Wed Sep  7 09:53:09 2016-date Wed Sep  7 09:50:27 2016; exit: 0 0 256 0 0
25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT Step0-PASSED Step1-PASSED Step2-FAILED Step3-NOTRUN Step4-NOTRUN  - time date Wed Sep  7 10:04:33 2016-date Wed Sep  7 09:50:28 2016; exit: 0 0 256 0 0
135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS Step0-PASSED Step1-PASSED Step2-PASSED  - time date Wed Sep  7 10:06:01 2016-date Wed Sep  7 09:51:01 2016; exit: 0 0 0
136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2 Step0-PASSED Step1-PASSED Step2-FAILED Step3-NOTRUN  - time date Wed Sep  7 10:02:52 2016-date Wed Sep  7 09:53:14 2016; exit: 0 0 256 0
140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI Step0-PASSED Step1-PASSED Step2-PASSED  - time date Wed Sep  7 10:06:33 2016-date Wed Sep  7 09:56:30 2016; exit: 0 0 0
1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Wed Sep  7 10:18:58 2016-date Wed Sep  7 10:03:01 2016; exit: 0 0 0 0
1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT Step0-PASSED Step1-FAILED Step2-NOTRUN Step3-NOTRUN Step4-NOTRUN  - time date Wed Sep  7 10:05:01 2016-date Wed Sep  7 10:04:36 2016; exit: 0 256 0 0 0
1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5 Step0-PASSED Step1-FAILED Step2-NOTRUN Step3-NOTRUN Step4-NOTRUN Step5-NOTRUN Step6-NOTRUN Step7-NOTRUN  - time date Wed Sep  7 10:05:37 2016-date Wed Sep  7 10:05:09 2016; exit: 0 256 0 0 0 0 0 0
10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Wed Sep  7 10:16:57 2016-date Wed Sep  7 10:05:43 2016; exit: 0 0 0 0
10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Wed Sep  7 10:29:26 2016-date Wed Sep  7 10:06:04 2016; exit: 0 0 0 0
10424.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D1_GenSimFull+DigiFull_2023D1+RecoFullGlobal_2023D1+HARVESTFullGlobal_2023D1 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Wed Sep  7 10:27:59 2016-date Wed Sep  7 10:06:39 2016; exit: 0 0 0 0
11224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D3_GenSimFull+DigiFull_2023D3+RecoFullGlobal_2023D3+HARVESTFullGlobal_2023D3 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Wed Sep  7 10:52:22 2016-date Wed Sep  7 10:17:02 2016; exit: 0 0 0 0
14 11 7 5 0 0 0 0 tests passed, 0 3 3 0 0 0 0 0 failed

```

**Comment by ghellwig** [comment]
@mmusich sorry, my bad

File: DQMOffline/Alignment/python/ALCARECOMuAlDQM_cff.py [hunk 1]
```diff
@@ -233,4 +291,4 @@
 #)
 #ALCARECOMuAlZMuMuDQM = cms.Sequence( ALCARECOMuAlZMuMuTrackingDQM + ALCARECOMuAlZMuMuMuAlDQM + ALCARECOMuAlZMuMuHLTDQM)
 ALCARECOMuAlZMuMuDQM = cms.Sequence( ALCARECOMuAlZMuMuTrackingDQM + ALCARECOMuAlZMuMuMuAlDQM)
-
+ALCARECOMuAlZMuMuGeneralTracksDQM = cms.Sequence( ALCARECOMuAlZMuMuGeneralTracksTrackingDQM + ALCARECOMuAlZMuMuGeneralTracksDQM)
```
Comment (mmusich) [inline DQMOffline/Alignment/python/ALCARECOMuAlDQM_cff.py]: @lpernie this line cannot be correct:
```
ALCARECOMuAlZMuMuGeneralTracksDQM = cms.Sequence( ALCARECOMuAlZMuMuGeneralTracksTrackingDQM 
+ 
ALCARECOMuAlZMuMuGeneralTracksDQM)
```
please rename the sequence to something else than the clone of `_MuonAl.muonAlignment.clone`

Comment (lpernie) [inline DQMOffline/Alignment/python/ALCARECOMuAlDQM_cff.py]: You are right, I just fixed it. Thanks!

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@lpernie would you please cherry-pick https://github.com/cms-sw/cmssw/commit/1de98b82727a96af29fdf789cd45cec92fc1f55d ?
This should fix the remaining failures

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@fabozzi, this PR fixes the behavior observed by @MengqingWu in the full sim cosmics wf 1307.0. It would be good to have it merged in pre12 to avoid patching again the matrix to submit the pre-release relvals

**Comment by fabozzi** [comment]
+1

**Review by dmitrijus — COMMENTED** [review]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
@mmusich - I agree - would be helpful if the PR description mentioned this useful bug fix...

**Pull request merged**